### DISCUSSION TITLE
Allow basic editing of vectors

### DISCUSSION THREAD

**Discussion by jamesvrt** [opengeos/leafmap#178] [Ideas]
I'm interested in using leafmap to annotate satellite imagery from within notebooks. Currently I can use `m.user_rois` to access drawn features which allows for a basic way to do this. However, I would like to be able to edit an existing vector dataset imported to the leafmap map. Some functionality that would be a good start:

- Add features to the vector dataset
- Fill out attributes for a selected feature
- Delete existing features
- Modify existing attribute fields for selected features

I've seen from @giswqs reply [here](https://github.com/jupyter-widgets/ipyleaflet/issues/599#issuecomment-629817042) that some of this functionality is available in geemap, but it doesn't appear it is possible in leafmap.

I can see that ipyleaflet [supports some basic polygon editing](https://ipyleaflet.readthedocs.io/en/latest/api_reference/polygon.html#editable-polygon) but I don't think this would let me do what I want.

**Comment by giswqs** [comment]
Good suggestion. This would be a nice feature to add. The functionality available in geemap is not really editing functionality. It only allows converting vector data to geojson format and then load it onto the map. It does not have functionality for editing geojson. 

I just explored the ipyleaflet DrawControl a little bit and found that it is possible to edit the `data` attribute. Basically, you can read a list features from a GeoJSON and assign it to the `data` attribute of the DrawControl. Then you can use the editing tool to edit shapes. Modifying attributes interactively might be possible as well, but it needs more work. I will see if I can implement it next week. 

![draw_control](https://user-images.githubusercontent.com/5016453/149437382-5cd069cc-5edc-4752-8c38-809bbe953401.gif)
